runestone/usageAssignment/__init__.py

```diff
@@ -57,10 +57,18 @@ def visit_ua_node(self,node):
         chapter_data = None
 
     s = ""
+    chapters_and_subchapters = {}
     if chapter_data and course_name:
-        for d in chapter_data:
+        for d in chapter_data: # Set up Chapter-Subchs dictionary
             ch_name, sub_chs = d['ch'], d['sub_chs']
-            s += '<div class="panel-heading">'
+            if d['ch'] not in chapters_and_subchapters:
+                chapters_and_subchapters[d['ch']] = [x for x in d['sub_chs']]
+            else:
+                for subch in d['sub_chs']:
+                    chapters_and_subchapters[d['ch']].append(subch)
+
+        for ch_name,sub_chs in chapters_and_subchapters.items():
```

Comment:
I can't imagine why I didn't think of that, that's embarrassing. It gave me the right order on first build because not-quite-coincidence, but yes, I do.